### ISSUE TITLE
[ENH] Adding `windows_identical` to `DirectReductionForecaster`

### ISSUE THREAD

**Issue by hliebert** [sktime/sktime#6545]
Due to the issues outlined in #5736 and #3224, we have been using the new reduction forecaster classes at work, in particular `DirectReductionForecaster`. So far, this has been working great for us.

One limitation of the new framework is that the windows used to train the estimator in `DirectReductionForecaster` are currently determined by the maximum forecasting horizon. We have found that this is a problem in our use case, as we are losing too much recent data in training. In the old reduction framework, there was a parameter `windows_identical` that allowed using horizon-specific windows in training. Having this feature in the new class would be extremely helpful.

**Comment by hliebert** [comment]
Thanks for the quick response. Below is some code that illustrates the model. In the example, the data is available until 1962. We would like data up to 1961 to be used to fit the model for horizon 1. Instead, data up to 1960 is used to fit the model for both horizons.

```python
import numpy as np
from sklearn.dummy import DummyRegressor
from sktime.datasets import load_longley
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import DirectTabularRegressionForecaster
from sktime.forecasting.compose._reduce import DirectReductionForecaster

y, X = load_longley()
horizon = ForecastingHorizon(np.arange(1, 3), is_relative=True)


# special sklearn model in order to inspect inputs into model methods
class DummyRegressorWithPrintout(DummyRegressor):
    """Sklearn dummy model that would print out input data before fitting and predicting"""
    def fit(self, X, y):
        print("## fit ")
        print("### ---------------- X -------------------")
        print(f"```\n{X}\n```")
        print("### ---------------- y -------------------")
        print(f"```\n{y}\n```")
        m = super().fit(X, y)
        return m

    def predict(self, X):
        print("## predict")
        print("### ---------------- X -------------------")
        print(f"```\n{X}\n```")
        return super().predict(X)
estimator = DummyRegressorWithPrintout()


# The data for 1961 should be used to train horizon 1, but instead only data up to 1960 is used for both horizons
print(y)
example_identical = DirectReductionForecaster(DummyRegressorWithPrintout(), X_treatment="shifted", window_length=4,)
example_identical.fit(y, fh=horizon)

# DirectTabularRegressionForecaster uses horizon specific data (but has other issues)
# example_non_identical = DirectTabularRegressionForecaster(DummyRegressorWithPrintout(), window_length=4, windows_identical=False)
# example_non_identical.fit(y, fh=horizon)
```

**Comment by fkiraly** [comment]
Thanks, I get it now. Are you open to contribute this? Indeed this would be a helpful addition.

The trade-off that identical windows may increase performance in case of reduction to multioutput models should be noted, and the current behaviour should be the default due to deprecation policies if we add the parameter, although it is worth considering what the default should be mid-term.

Also, of course, very happy to help with navigating the current code.

**Comment by hliebert** [comment]
Sure, I'll try to carve out some time to work on this. I'd appreciate some help navigating the current code.

**Comment by fkiraly** [comment]
sure - ask away if you would like to know anything. Also happy to give a walkthrough in one of the meet-ups.

**Comment by hliebert** [comment]
Implemented in #6650.
